feat: at_repl 2.1.0

JeremyTubongbanua opened the pull request:
**- What I did**
- NEW FEATURE: Added `--key-file <keyfilepath>` option (`-k` for short)
- NEW FEATURE: New `--version`
- Renamed `--rootUrl` to `--rootDomain`, kept `--rootUrl` as an alias
- Now uses AtRootDomain (supports proxy servers)

Other stuff
- Upgraded version to `2.1.0`
- Now uses `at_client ^3.8.0`
- Refactored the use of futures everywhere for better usability of functions

**- How I did it**

With my best friend Codex

**- How to verify it**

Sample execution

<img width="627" height="561" alt="image" src="https://github.com/user-attachments/assets/e6f62bcf-3b44-440d-bb6a-94a5dd173f33" />

**- Description for the changelog**
feat!: at_repl 2.1.0

JeremyTubongbanua committed:
feat: `--keys|-k` option

JeremyTubongbanua committed:
chore: update at_client to ^3.8.0

JeremyTubongbanua committed:
feat: use futures, change rootUrl to root-domain, and add --version, update README

JeremyTubongbanua committed:
chore: update at_repl version to 2.1.0 because rootUrl-->root-domain is a breaking change

gkc reviewed (changes requested):
- Versions in CHANGELOG and pubspec.yaml do not match
- Please keep --rootUrl as an alias

JeremyTubongbanua committed:
chore: format the parser flags, remove `-V`, remove `--keys`, unhide `--key-file` rename it to `--key-file`

JeremyTubongbanua committed:
chore: abbr 'k' and `--key-file`

JeremyTubongbanua committed:
docs: fix up CHANGELOG.md

JeremyTubongbanua committed:
chore: remove inaccurate print statement

JeremyTubongbanua committed:
chore: print the version more simply

JeremyTubongbanua committed:
docs: at_repl/CHANGELOG.md

JeremyTubongbanua commented:
@gkc 

Updated the CHANGELOG
`--rootUrl` is now an alias

gkc commented on packages/at_repl/bin/at_repl.dart:
```diff
@@ -1,53 +1,77 @@
+      defaultsTo: 'root.atsign.org:64',
+      help: 'The root domain (optionally host:port) to connect to',
+      aliases: ['rootUrl'])
+    ..addOption('key-file',
```
why not keep --keys-file as an alias?

JeremyTubongbanua replied:
https://github.com/atsign-foundation/at_tools/pull/488/commits/107832b4352adcce54c5599801c32373bafbfb64

gkc commented on packages/at_repl/bin/at_repl.dart:
```diff
@@ -1,53 +1,77 @@
-  if (atSign == null || atSign.isEmpty) {
+  final String? atSignArg = results['atSign'] as String?;
+  if (atSignArg == null || atSignArg.trim().isEmpty) {
     print('Error: atSign is required');
```
it's always required because you are setting `mandatory:true` when adding the option

JeremyTubongbanua replied:
https://github.com/atsign-foundation/at_tools/pull/488/commits/a2b259a337544fbc2338fd94e6733ad8312dcc3b

gkc commented on packages/at_repl/bin/at_repl.dart:
```diff
@@ -1,53 +1,77 @@
+  try {
+    atSign = AtUtils.fixAtSign(atSignArg);
+  } catch (e) {
+    print('Error: $e');
```
Use the `Atsign` class and the String.toAtsign() method

JeremyTubongbanua replied:
https://github.com/atsign-foundation/at_tools/pull/488/commits/a2b259a337544fbc2338fd94e6733ad8312dcc3b

gkc commented on packages/at_repl/bin/at_repl.dart:
```diff
@@ -1,53 +1,77 @@
-  final String? keysFile = results['keys-file'] as String?;
+  String rootUrl = results['root-domain'] as String;
+  final String? keysPath = results.wasParsed('keys')
+      ? results['keys'] as String?
```
You are not adding `keys` as an option so `keys` will never be parsed

JeremyTubongbanua replied:
It is now parsing `key-file` https://github.com/atsign-foundation/at_tools/pull/488/commits/34a611cc48dbc278277c7b6076a4b560b425589a

gkc reviewed (changes requested):
See comments

JeremyTubongbanua committed:
fix: --key-file flag causing error

JeremyTubongbanua committed:
merge: remote-tracking branch 'upstream/trunk' into jt/at_repl

JeremyTubongbanua committed:
chore: add .DS_Store to .gitignore

JeremyTubongbanua committed:
merge: remote-tracking branch 'upstream/trunk' into jt/at_repl

JeremyTubongbanua committed:
feat: add keys-file and keys as aliases to -k option

dependabot[bot] committed:
chore(deps): bump github/codeql-action in the github-actions group

Bumps the github-actions group with 1 update: [github/codeql-action](https://github.com/github/codeql-action).


Updates `github/codeql-action` from 4.32.2 to 4.32.3
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/45cbd0c69e560cd9e7cd7f8c32362050c9b7ded2...9e907b5e64f6b83e7804b09294d44122997950d6)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.32.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action in the github-actions group

Bumps the github-actions group with 1 update: [github/codeql-action](https://github.com/github/codeql-action).


Updates `github/codeql-action` from 4.32.3 to 4.32.4
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/9e907b5e64f6b83e7804b09294d44122997950d6...89a39a4e59826350b863aa6b6252a07ad50cf83e)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.32.4
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump actions/upload-artifact in the github-actions group

Bumps the github-actions group with 1 update: [actions/upload-artifact](https://github.com/actions/upload-artifact).


Updates `actions/upload-artifact` from 6.0.0 to 7.0.0
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/b7c566a772e6b6bfb58ed0dc250532a479d7789f...bbbca2ddaa5d8feaa63e36b76fdaad77386f024f)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: 7.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action in the github-actions group

Bumps the github-actions group with 1 update: [github/codeql-action](https://github.com/github/codeql-action).


Updates `github/codeql-action` from 4.32.4 to 4.32.5
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/89a39a4e59826350b863aa6b6252a07ad50cf83e...c793b717bc78562f491db7b0e93a3a178b099162)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.32.5
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action in the github-actions group

Bumps the github-actions group with 1 update: [github/codeql-action](https://github.com/github/codeql-action).


Updates `github/codeql-action` from 4.32.5 to 4.32.6
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/c793b717bc78562f491db7b0e93a3a178b099162...0d579ffd059c29b07949a3cce3983f0780820c98)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.32.6
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action in the github-actions group

Bumps the github-actions group with 1 update: [github/codeql-action](https://github.com/github/codeql-action).


Updates `github/codeql-action` from 4.32.6 to 4.33.0
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/0d579ffd059c29b07949a3cce3983f0780820c98...b1bff81932f5cdfc8695c7752dcee935dcd061c8)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.33.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action in the github-actions group

Bumps the github-actions group with 1 update: [github/codeql-action](https://github.com/github/codeql-action).


Updates `github/codeql-action` from 4.33.0 to 4.34.1
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/b1bff81932f5cdfc8695c7752dcee935dcd061c8...38697555549f1db7851b81482ff19f1fa5c4fedc)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.34.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump dart-lang/setup-dart in the github-actions group

Bumps the github-actions group with 1 update: [dart-lang/setup-dart](https://github.com/dart-lang/setup-dart).


Updates `dart-lang/setup-dart` from 1.7.1 to 1.7.2
- [Release notes](https://github.com/dart-lang/setup-dart/releases)
- [Changelog](https://github.com/dart-lang/setup-dart/blob/main/CHANGELOG.md)
- [Commits](https://github.com/dart-lang/setup-dart/compare/e51d8e571e22473a2ddebf0ef8a2123f0ab2c02c...65eb853c7ba17dde3be364c3d2858773e7144260)

---
updated-dependencies:
- dependency-name: dart-lang/setup-dart
  dependency-version: 1.7.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump google/osv-scanner-action in the github-actions group

Bumps the github-actions group with 1 update: [google/osv-scanner-action](https://github.com/google/osv-scanner-action).


Updates `google/osv-scanner-action` from 2.3.3 to 2.3.5
- [Release notes](https://github.com/google/osv-scanner-action/releases)
- [Commits](https://github.com/google/osv-scanner-action/compare/c5996e0193a3df57d695c1b8a1dec2a4c62e8730...c51854704019a247608d928f370c98740469d4b5)

---
updated-dependencies:
- dependency-name: google/osv-scanner-action
  dependency-version: 2.3.5
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump github/codeql-action in the github-actions group

Bumps the github-actions group with 1 update: [github/codeql-action](https://github.com/github/codeql-action).


Updates `github/codeql-action` from 4.34.1 to 4.35.1
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/38697555549f1db7851b81482ff19f1fa5c4fedc...c10b8064de6f491fea524254123dbe5e09572f13)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-version: 4.35.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

cpswan committed:
build(deps): Add dependency cooldowns

dependabot[bot] committed:
chore(deps): bump actions/upload-artifact in the github-actions group

Bumps the github-actions group with 1 update: [actions/upload-artifact](https://github.com/actions/upload-artifact).


Updates `actions/upload-artifact` from 7.0.0 to 7.0.1
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/bbbca2ddaa5d8feaa63e36b76fdaad77386f024f...043fb46d1a93c77aae656e7c1c64a875d1fc6a0a)

---
updated-dependencies:
- dependency-name: actions/upload-artifact
  dependency-version: 7.0.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: github-actions
...

Signed-off-by: dependabot[bot] <support@github.com>

JeremyTubongbanua committed:
feat: use AtRootDomain

JeremyTubongbanua committed:
docs: CHANGELOG

JeremyTubongbanua committed:
chore: root domain aliases

JeremyTubongbanua committed:
feat: use Atsign typedef

JeremyTubongbanua committed:
fix: breaking change

JeremyTubongbanua committed:
chore: remove -V ?

JeremyTubongbanua committed:
feat: update version

JeremyTubongbanua committed:
fix: make version printing cleaner code

JeremyTubongbanua committed:
feat: `-r` abbr for root domain

JeremyTubongbanua committed:
Merge remote-tracking branch 'upstream/trunk' into jt/at_repl
